gdnative-core/src/export/signal.rs

```diff
@@ -0,0 +1,46 @@
+use crate::core_types::{GodotString, Variant};
+use crate::export::{ClassBuilder, ExportInfo, NativeClass, PropertyUsage};
+
+pub struct SignalBuilder<'a, C> {
+    class_builder: &'a ClassBuilder<C>,
+    name: GodotString,
+    args: Vec<SignalArgument>,
+}
+
+impl<'a, C: NativeClass> SignalBuilder<'a, C> {
+    pub(super) fn new(class_builder: &'a ClassBuilder<C>, name: &str) -> Self {
+        Self {
+            class_builder,
+            name: GodotString::from(name),
+            args: vec![],
+        }
+    }
```

Comment:
Perhaps we should just make `name` a type parameter then, if we just convert it to `GodotString`? Same with `ClassBuilder::signal`.

Reply:
You probably mean `name: impl Into<GodotString>`? Yes, that's a good idea.